utils/terrascript_client.py

```diff
@@ -765,6 +768,95 @@ def populate_tf_resource_dynamodb(self, resource, namespace_info):
         for tf_resource in tf_resources:
             self.add_resource(account, tf_resource)
 
+    def populate_tf_resource_ecr(self, resource, namespace_info):
+        account, identifier, common_values, \
+            output_prefix, output_resource_name = \
+            self.init_values(resource, namespace_info)
+
+        tf_resources = []
+        self.init_common_outputs(tf_resources, namespace_info,
+                                 output_prefix, output_resource_name)
+
+        # ecr repository
+        # Terraform resource reference:
+        # https://www.terraform.io/docs/providers/aws/r/ecr_repository.html
+        values = {}
+        values['name'] = identifier
+        values['tags'] = common_values['tags']
+
+        ecr_tf_resource = aws_ecr_repository(identifier, **values)
+        tf_resources.append(ecr_tf_resource)
+        output_name = output_prefix + '[url]'
+        output_value = '${' + ecr_tf_resource.fullname + '.repository_url}'
+        tf_resources.append(output(output_name, value=output_value))
+        region = common_values['region'] or self.default_regions[account]
+        output_name = output_prefix + '[aws_region]'
+        tf_resources.append(output(output_name, value=region))
+
+        # iam resources
+        # Terraform resource reference:
+        # https://www.terraform.io/docs/providers/aws/r/iam_access_key.html
+
+        # iam user for repository
+        values = {}
+        values['name'] = identifier
+        values['tags'] = common_values['tags']
+        values['depends_on'] = [ecr_tf_resource]
+        user_tf_resource = aws_iam_user(identifier, **values)
+        tf_resources.append(user_tf_resource)
+
+        # iam access key for user
+        tf_resources.extend(
+            self.get_tf_iam_access_key(
+                user_tf_resource, identifier, output_prefix))
+
+        # iam user policy for bucket
+        values = {}
+        values['user'] = identifier
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "ListImagesInRepository",
+                    "Effect": "Allow",
+                    "Action": ["ecr:ListImages"],
+                    "Resource": "${" + ecr_tf_resource.fullname + ".arn}"
+                },
+                {
+                    "Sid": "GetAuthorizationToken",
+                    "Effect": "Allow",
+                    "Action": ["ecr:GetAuthorizationToken"],
+                    "Resource": "*"
```

Comment:
should the resource really be `all` here?

Reply:
iiuc, you get a token for the registry (one per region), and not per repository. so i think that `all` makes sense...